### ISSUE TITLE
docker-run --tty does not work

### ISSUE THREAD

**Issue by hunterboerner** [mitchellh/vagrant#7597]
Verified on Mac & Linux. Vagrant master branch.

The `--tty` flag does not work no the docker-run command (possibly others).

# Example Vagrantfile

```ruby
ENV['VAGRANT_DEFAULT_PROVIDER'] ||= 'docker'

Vagrant.configure("2") do |config|
  config.vm.synced_folder ".", "/vagrant", disabled: true

  config.vm.provider "docker" do |d|
    d.remains_running = false
    d.force_host_vm = false # I do not use a host VM with Docker. I have not test this with a host VM.
    d.image = "ubuntu"
  end
end
```

# Example output

```
⋊> ~/c/t/vagrant_docker vagrant docker-run --tty -- echo what                                                                                                                     10:34:25
==> default: Creating the container...
    default:   Name: vagrant_docker_default_1468942469_1468942469
    default:  Image: ubuntu
    default:    Cmd: echo what
    default:
    default: Container is starting. Output will stream in below...
    default:
the input device is not a TTYA Docker command executed by Vagrant didn't complete successfully!
The command run along with the output from the command is shown
below.

Command: ["docker", "run", "--name", "vagrant_docker_default_1468942469_1468942469", "-i", "-t", "--rm=true", "ubuntu", "echo", "what", {:notify=>[:stdout, :stderr]}]

Stderr: the input device is not a TTY


Stdout:
```

# Explanation

The `--tty` flag adds `-i` and `-t` to the docker arg list. `-i` requires STDIN to be a TTY. However, Vagrant runs commands in an environment where STDIN cannot be a TTY. See: https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/util/subprocess.rb#L73 and https://github.com/enkessler/childprocess/blob/ddde6c9dbf7146d9693410c0138807b4f951f816/lib/childprocess/unix/fork_exec_process.rb#L35-L38. I propose removing the `-i` flag (https://github.com/mitchellh/vagrant/blob/master/plugins/providers/docker/driver.rb#L51) from the docker command (and maybe provide a separate flag for `-i` users).

**Comment by sethvargo** [comment]
Hi there,

Thank you for opening an issue. However, there is not enough information here to reproduce or provide a recommendation. When you first open an issue, a template is pre-filled in the box. That template provides instructions for getting your host and guest information, the version of Vagrant, and the debug output. Without that information, it is very challenging for us to assist you.

Please supply that information either by updating this ticket or by closing this ticket and opening a new one. Specifically, we need a Vagrantfile and Vagrant command to reproduce this issue as well as the debug output. I appreciate that you've tracked down where you think the issue may lie, but it's important for us to have adequate reproduction steps. Thank you! 😄

**Comment by hunterboerner** [comment]
@sethvargo I've updated the issue. Is this debug output necessary for this issue?

**Comment by hunterboerner** [comment]
@sethvargo any word on this? The fix should be fairly easy to implement.

**Comment by sethvargo** [comment]
Hi @hunterboerner 

We are actively triaging issues, and we will get to this as soon as possible. If you'd like to prioritize a fix, submitting a patch for review is the best approach. Thanks!

**Comment by chrisroberts** [comment]
Fixed via #7688
